engine/session/gitcredential.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+option go_package = "session";
+
+service GitCredential {
+  rpc GetCredential(GitCredentialRequest) returns (GitCredentialResponse);
+}
+
+message GitCredentialRequest {
+  string protocol = 1;
+  string host = 2;
+  string path = 3; // optional
+}
+
+message GitCredentialResponse {
+  oneof result {
+    CredentialInfo credential = 1;
+    ErrorInfo error = 2;
```

Comment:
nit (can be changed later): I **think** the gRPC way of doing this is to return an actual error from the function, rather returning `nil` as an error with a custom `error` field in the response